backend/src/packages/auth/auth.controller.ts

```diff
@@ -75,6 +94,28 @@ class AuthController extends BaseController {
       payload: await this.authService.signUp(options.body),
     };
   }
+
+  private async signIn(
+    options: APIHandlerOptions<{
+      body: UserSignInRequestDto;
+    }>,
+  ): Promise<APIHandlerResponse> {
+    const user = await this.authService.verifyLoginCredentials(options.body);
+
+    if (!user) {
+      return {
+        status: HTTPCode.UNAUTHORIZED,
+        payload: { message: 'Unauthorized' },
```

Comment:
I think status should be NOT_FOUND with 404 code, because in fact verifyLoginCredentials checks, if the user with this data exists. Also the message will be transffered to frontend and as I understand will be displayed to the client, so it's better to make it more detailed, for instance 'User with these credentials was not found'

Reply:
Since we are dealing with authentication and user login, if the user's credentials are invalid, returning an "Unauthorized" error makes more sense. This clearly communicates that the authentication process failed due to incorrect credentials, and the user should try again with valid credentials. And "Not Found" error is used when the requested resource is not found on the server. Question is then, at what point it is best to return "Unauthorized" - during password checking?

Reply:
In previous projects (for example backend structure) usually unauthorised was used in cases when token is wrong or client don't have enough permissions, while in login in case of not finding user with the same email was something like "User no found" exception

Reply:
Ok, I added it!